packages/grid/x-grid/src/themeAugmentation/props.ts

```diff
@@ -1,9 +1,21 @@
+import { ComponentsOverrides, ComponentsProps } from '@material-ui/core/styles';
 import { DataGridProProps } from '../DataGridProProps';
 
-export interface DataGridProComponentProps {
-  MuiDataGridPro: DataGridProProps;
+export interface DataGridProComponentsPropsList {
+  MuiDataGrid: DataGridProProps;
+}
+
+export interface DataGridProComponents {
+  MuiDataGrid?: {
+    defaultProps?: ComponentsProps['MuiDataGrid'];
+    styleOverrides?: ComponentsOverrides['MuiDataGrid'];
+  };
 }
 
 declare module '@material-ui/core/styles' {
-  interface ComponentsPropsList extends DataGridProComponentProps {}
+  interface ComponentsPropsList extends DataGridProComponentsPropsList {}
+}
+
+declare module '@material-ui/core/styles/components' {
```

Comment:
It's a private path:

```suggestion
declare module '@material-ui/core/styles' {
```

Reply:
The `Components` interface is not exported. We will be able to remove the private path when @mui/material 5.0.0-rc.0 became the minimum version. The RC includes https://github.com/mui-org/material-ui/pull/28078.